table/server/underdb/hive/src/main/java/alluxio/table/under/hive/HiveDatabase.java

```diff
@@ -265,4 +263,19 @@ HiveMetaStoreClient getHive() throws IOException {
       Thread.currentThread().setContextClassLoader(currentClassLoader);
     }
   }
+
+  HiveMetaStoreClient getHive() throws IOException {
+    if (mHive != null) {
+      try {
+        mHive.getAllDatabases(); // test the connection
```

Comment:
Is there a better way to detect if it is connected? Doing an extra RPC sounds inefficient.

An alternative can be to just do the desired operations with the existing cached client, and if it fails, try again with a re-connected client.

Reply:
Unfortunately, `isConnected` is a private variable in the client. 
I thought about doing the second approach, but there is no way to encapsulate it in the `getHive` method though? That means for each call site of `getHive` we need to do this exception handling?

Reply:
What was the scenario where this would happen? (I had never seen a client get disconnected)

I don't think calling an extra RPC just to check the connection state is what we want to do. Maybe there is a simple retry utility we can write?